composer.json

```diff
@@ -18,18 +18,18 @@
         }
     ],
     "require": {
-        "php": "^7.4 || ^8.0",
-        "doctrine/dbal": "^2.8 || ^3.0",
+        "php": "^8.1",
+        "doctrine/dbal": "^4.0",
```

Comment:
Not sure this is going to fly… ideally you should support `^3.0 || ^4.0` so that people can upgrade this library, deploy, and then upgrade `doctrine/dbal`

Reply:
@greg0ire the changes between v3 & v4 are too big: https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#upgrade-to-40

That's why my suggestion would be to maintain a new version that only supports dbal v4: https://github.com/ramsey/uuid-doctrine/issues/246

Reply:
I think for most changes, there is a forward compatibility layer, so it's likely that if you bump the DBAL requirement to  a high enough 3.x version, you will be able to do some of the changes in this PR.